build-prisma-fmt/flake.nix

```diff
@@ -28,7 +28,7 @@
           checkPhase = "echo 'checkPhase: do nothing'"; # TODO: we should check we have a non-empty .wasm file here
           name = "prisma-fmt-wasm";
           src = ./.;
-          cargoSha256 = "sha256-zu+ELa7aEdLgq4ycImLQvqlG1MCa6GrkVCLZuDYJF08=";
+          cargoSha256 = "sha256-fj0dmGS6/RFkOugH9zpTmlU+jXq6+EYoOGXeA9tr/Aw=";
```

Comment:
This is the only value that does not come as an external input. Where does this come from? Is this just a sha of the `Cargo.lock` or something? What is this important for?

Reply:
I dug a bit deeper since it made me wonder: the hash is certainly necessary for projects without a Cargo.lock, but for those with one, it should be possible to rely on it.

> buildRustPackage requires either the cargoSha256 or the cargoHash attribute which is computed over all crate sources of this package.

(source: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md)

But judging from https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#importing-a-cargolock-file-importing-a-cargolock-file, we could do with the Cargo.lock only, I'll try that monday.

Reply:
My other thought was that this might be a sha over what Cargo.lock tells the project to use, so knowing what we have in there might be enough to reproduce this one - and as the Cargo.lock changes only rely on the Engine hash - we could possibly indeed skip this commit as everything depends on the Engine hash which is input for the update.

Reply:
The `Cargo.lock` will do also include dependencies of engines (anything non-breaking according the the version strings in the Cargo.toml) when they update. The `Cargo.lock` of prisma-engines isn't taken into account here (they never are for dependencies).

Reply:
No, we use the default, meaning if you write `serde = "1.0.0"` (or `"1"`, or `"1.2"`, equivalently), a `cargo update` will update the dependency as high as it can, just not the major version.

Reply:
Oh ok, that explains a few things.

Do we ship a `Cargo.lock` with the engine files somehow that we could use to get a deterministic setup of dependencies to build the wasm version that uses the same as the main builds?

Reply:
So it's not how cargo handles things normally, you have one lockfile at the top of your project that locks your deliverable (99% of the time an executable, but in this case the wasm module). The dependencies don't have a lockfile, or if they have one it's ignored, and from this project's perspective, engines is a dependency (through `datamodel`).

Reply:
The way to use the engines lockfile for this wasm crate would be having the crate in the engines workspace (repo), but that turned out to make the buildkite build pipeline break, so I don't have a perfect solution here.